examples/cms/src/data.ts

```diff
@@ -0,0 +1,126 @@
+import { CollectionItemData, ManagedCollectionField, framer } from "framer-plugin"
+import { generateHash, isNotNull, slugify } from "./utils"
+
+// Plugin keys
+export const PLUGIN_PREFIX = "cms_starter"
+export const LOCAL_STORAGE_LAST_LAUNCH_KEY = `${PLUGIN_PREFIX}.lastLaunched`
+
+export const PLUGIN_COLLECTION_SYNC_REFERENCE_KEY = `collectionSyncReference`
+export const PLUGIN_COLLECTION_SYNC_SLUG_KEY = `collectionSyncSlug`
+
+export type ManagedCollectionFieldType = ManagedCollectionField["type"]
+
+type DataSourceField = {
+    type: ManagedCollectionFieldType
+}
+
+export interface DataSource {
+    id: string
+    fields: Record<string, DataSourceField>
+    items: Record<string, unknown>[]
+}
+
+export interface FieldConfig {
+    name: string
+    isNew: boolean
+    field: ManagedCollectionField | null
+}
+
+export async function listDataSourcesIds(): Promise<string[]> {
+    return ["articles", "categories"]
+}
+
+export async function getDataSources(collection: string): Promise<DataSource> {
+    return await fetch(`/datasources/${collection}.json`).then(res => res.json())
+}
+
+export function computeFieldConfigs(existingFields: ManagedCollectionField[], dataSource: DataSource) {
+    const result: FieldConfig[] = []
+    const fields = dataSource.fields
+
+    for (const [name, fieldType] of Object.entries(fields)) {
+        const fieldId = generateHash(name)
+        let newField: ManagedCollectionField | null = null
+
+        const existingField = existingFields.find(field => field.id === fieldId)
+        if (existingField) {
+            newField = existingField
+        } else {
+            newField = {
+                id: fieldId,
+                name,
+                type: fieldType.type,
+                userEditable: false,
+            } as ManagedCollectionField
```

Comment:
Do we need to cast this?

Reply:
Yes, because `type` could be `"enum" | "collectionReference" | "multiCollectionReference"` and then the field schema changes (either requiring `collectionId` or `cases`).